var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -274,37 +272,38 @@ def cmake_args(self):
     #
     # export SPACK_USER_CACHE_PATH=/tmp/spack
     # export SPACK_DISABLE_LOCAL_CONFIG=true
-    def test(self):
-        if not self.spec.satisfies("@develop") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
-
-        tests = [
-            ["NlpMdsEx1.exe", "400", "100", "0", "-selfcheck"],
-            ["NlpMdsEx1.exe", "400", "100", "1", "-selfcheck"],
-            ["NlpMdsEx1.exe", "400", "100", "0", "-empty_sp_row", "-selfcheck"],
+
+    def run_hiop(self, raja):
+
+        if raja:
+            exName = "NlpMdsEx1Raja.exe"
+        else:
+            exName = "NlpMdsEx1.exe"
+
+        exe = os.path.join(self.prefix.bin, exName)
+        if not os.path.exists(exe):
+            raise SkipTest(f"{exName} does not exist in version {self.version}")
+
+        options = [
+            ["400", "100", "0", "-selfcheck"],
+            ["400", "100", "1", "-selfcheck"],
+            ["400", "100", "0", "-empty_sp_row", "-selfcheck"],
         ]
 
-        if "+raja" in self.spec:
-            tests.extend(
-                [
-                    ["NlpMdsEx1Raja.exe", "400", "100", "0", "-selfcheck"],
-                    ["NlpMdsEx1Raja.exe", "400", "100", "1", "-selfcheck"],
-                    ["NlpMdsEx1Raja.exe", "400", "100", "0", "-empty_sp_row", "-selfcheck"],
-                ]
-            )
+        exe = which(exe)
 
-        for i, test in enumerate(tests):
-            exe = os.path.join(self.prefix.bin, test[0])
-            args = test[1:]
-            reason = 'test {0}: "{1}"'.format(i, " ".join(test))
-            self.run_test(
-                exe,
-                args,
-                [],
-                0,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
-                work_dir=self.prefix.bin,
-            )
+        for i, args in enumerate(options):
+            with test_part(
+                self, f"test_{exName}_{i+1}", purpose="{0} {1}".format(str(exe), " ".join(args))
```

Comment:
:thinking: I now regret suggesting inclusion of `{exe}` in the purpose as it is *very* messy with the entire path.  Better to just output the joining of the args.

```
 test: test_NlpMdsEx1.exe_2: $spack/opt/spack/linux-rhel8-x86_64_v3/gcc-12.1.1/hiop-1.0.3-vecd7zfgdeksyfqdlm32n5feamcehcpi/bin/NlpMdsEx1.exe 400 100 1 -selfcheck
 ```

There's also some redundancy with the output from Spack's `Executable` but the documentation for the examples seems pretty sparse so can't really get something more useful there.